Mathlib/Analysis/Convex/PathSegment.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2025 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Analysis.Convex.Basic
+import Mathlib.Topology.Connected.PathConnected
+
+/-!
+# Segment between 2 points as a bundled path
+
+In this file we define `Path.segment a b : Path a b`
+to be the path going from `a` to `b` along the straight segment with constant velocity `b - a`.
+
+We also prove basic properties of this construction,
+then use it to show that a nonempty convex set is path connected.
+-/
+
+open Set
+open scoped Convex
+
+variable {E : Type*} [AddCommGroup E] [Module ℝ E]
+  [TopologicalSpace E] [ContinuousAdd E] [ContinuousSMul ℝ E]
+
+namespace Path
+
+/-- The path from `a` to `b` going along the straight segment. -/
+@[simps]
+protected def segment (a b : E) : Path a b where
+  toFun t := AffineMap.lineMap a b (t : ℝ)
+  source' := by simp
+  target' := by simp
+
+@[simp]
+theorem range_segment (a b : E) : Set.range (Path.segment a b) = [a -[ℝ] b] := by
+  simp only [segment_eq_image_lineMap, image_eq_range, ← segment_apply]
+
+@[simp]
+theorem segment_same (a : E) : Path.segment a a = .refl a := by ext; simp
```

Comment:
I'd like another reviewer to comment on this name --- I'm not sure (i.e., I just don't know) if that's our mathlib convention.

Reply:
I copied from https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Convex/Segment.html#segment_same. Probably, this means that this one should be `protected`.